### PR TITLE
Fix warnings on cpp-package

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
Fix warnings, unused variables

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,7 +3,6 @@
 
 ## Checklist ##
 ### Essentials ###
-- [ ] Passed code style checking (`make lint`)
```
Comment (eric-haibin-lin) [inline .github/PULL_REQUEST_TEMPLATE.md]: Why is this removed?

Comment (larroy) [reply .github/PULL_REQUEST_TEMPLATE.md]: I've talked to a couple of people and we are checking this in ci, what's the point of running it locally?

Comment (szha) [reply .github/PULL_REQUEST_TEMPLATE.md]: Since there was a bottleneck for scheduling and running tasks on old CI, each failed attempt even for sanity test meant wasted time, which was the motivation to add it here. You can propose a change to this with the updated status of the CI and your motivation.

May I request that you put this proposal either in an email to dev-list, or in a separate PR, so that only changes that are relevant in the notes and PR title are included?

Comment (larroy) [reply .github/PULL_REQUEST_TEMPLATE.md]: @szha ok

File: cpp-package/include/mxnet-cpp/optimizer.hpp [hunk 2]
```diff
@@ -296,7 +296,7 @@ inline void AdamOptimizer::Update(int index, NDArray weight, NDArray grad) {
   CHECK_EQ(keys.size(), values.size());
 
   float lr = std::stof(params_["lr"]);
-  float wd = std::stof(params_["wd"]);
+  // float wd = std::stof(params_["wd"]);
```
Comment (szha) [inline cpp-package/include/mxnet-cpp/optimizer.hpp]: Should the commented code be removed?

**Comment by larroy** [comment]
@szha ping

**Comment by larroy** [comment]
Thankyou.

**Comment by szha** [comment]
Thanks for the fix.
